helix-rest/src/main/java/org/apache/helix/rest/server/resources/helix/waged/WagedInstanceAccessor.java

```diff
@@ -0,0 +1,50 @@
+package org.apache.helix.rest.server.resources.helix.waged;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import java.util.Collections;
+import java.util.Map;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.core.Response;
+
+import org.apache.helix.HelixAdmin;
+import org.apache.helix.HelixException;
+import org.apache.helix.rest.server.resources.helix.AbstractHelixResource;
+
+
+@Path("/clusters/{clusterId}/instances")
+public class WagedInstanceAccessor extends AbstractHelixResource {
+  @GET
+  @Path("{instanceName}/validate")
+  public Response validateInstance(@PathParam("clusterId") String clusterId,
+      @PathParam("instanceName") String instanceName) {
+    HelixAdmin admin = getHelixAdmin();
+    Map<String, Boolean> validationResultMap;
+    try {
+      validationResultMap = admin
+          .validateInstancesForWagedRebalance(clusterId, Collections.singletonList(instanceName));
+    } catch (HelixException e) {
```

Comment:
Are you referring to badRequest? Do you think we should make this an internal server error or a bad request on the client side? Because this is validation logic, if validation fails, then should we consider it a bad request?

Reply:
But what if Helix has OOM or other problem for the system? What I mean that is we keep this catch (HelixException) but also add another catch (Exception) for general exception. Otherwise, user may get confused about our response.

Reply:
Good suggestion. I can add a catch clause for Exception as well.